pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java

```diff
@@ -1085,4 +1099,29 @@ public void cleanup() {
       }
     }
   }
+
+  @FunctionalInterface
+  public interface ExceptionalRunnable {
+    void run()
+        throws Exception;
+  }
+
+  protected void runWithHelixMock(ExceptionalRunnable r) {
+    try (MockedStatic<HelixPropertyFactory> mock = Mockito.mockStatic(HelixPropertyFactory.class)) {
+
+      // mock helix method to disable slow, but useless, getCloudConfig() call
+      Mockito.when(HelixPropertyFactory.getCloudConfig(Mockito.anyString(), Mockito.anyString()))
```

Comment:
Hmm, this breaks the purpose of integration test, where it should mimic the real cluster scenario.
How long does this call take?

Reply:
That call always returns an empty CloudConfig in tests.
Please have a look at the attached flame graph.